Spitfire/RtcConductor.cpp

```diff
@@ -238,19 +249,18 @@ namespace Spitfire
 
 	void RtcConductor::DataChannelSendText(const std::string & label, const std::string & text)
 	{
-		auto observer = dataObservers.find(label);
-		if (observer != dataObservers.end()) {
-			observer->second->dataChannel->Send(webrtc::DataBuffer(text));
+		if (dataObservers.find(label) != dataObservers.end()) {
+			dataObservers[label]->dataChannel->Send(webrtc::DataBuffer(text));
 		}
 	}
 
 	RtcDataChannelInfo RtcConductor::GetDataChannelInfo(const std::string& label)
 	{
 		auto info = RtcDataChannelInfo();
-		const auto observer = dataObservers.find(label);
-		if (observer != dataObservers.end()) {
 
-			const auto dataChannel = observer->second->dataChannel;
+		if (dataObservers.find(label) != dataObservers.end()) {
+
+			const auto dataChannel = dataObservers[label]->dataChannel;
```

Comment:
Looking at this a bit more closely, if this particular function is one that needs to be as efficient as possible, then most likely the way the code was before is probably a bit more efficient because dataObservers.find(label) and dataObservers[label] both end up doing a lookup on the map, which if it is a std::map then it is a binary tree. If it is an std::unordered_map then it's a hash table (constant time lookup), so not as big deal in that case. However, the way the code was before it returns an iterator to the item in the map, so only a single lookup is done.